--- a/drivers/cpufreq/spacemit-cpufreq.c	2024-11-09 02:15:06.754542972 +0200
+++ b/drivers/cpufreq/spacemit-cpufreq.c	2024-11-09 02:29:29.080949897 +0200
@@ -49,7 +49,7 @@ of_hotplug_cooling_register(struct cpufr
 #define FILTER_POINTS_0			(135)
 #define FILTER_POINTS_1			(142)
 
-#define K1_MAX_FREQ_LIMITATION		(1600000)
+#define K1_MAX_FREQ_LIMITATION		(1800000)
 #define M1_MAX_FREQ_LIMITATION		(1800000)
 
 #endif
--- a/drivers/cpufreq/spacemit-cpufreq.c	2024-11-09 02:15:06.754542972 +0200
+++ b/drivers/cpufreq/spacemit-cpufreq.c	2024-11-09 02:27:55.048687992 +0200
@@ -435,9 +435,12 @@ int spacmeit_cpufreq_veritfy(struct cpuf
 	} else {
 		/* K1 */
 		/* only 1.6G allowed max */
-		policy->max = policy->max > K1_MAX_FREQ_LIMITATION ? K1_MAX_FREQ_LIMITATION : policy->max;
+		/* policy->max = policy->max > K1_MAX_FREQ_LIMITATION ? K1_MAX_FREQ_LIMITATION : policy->max;
+		 * cpufreq_verify_within_limits(policy, policy->cpuinfo.min_freq,
+		 * 			K1_MAX_FREQ_LIMITATION); */
+		/* can update to 1.8G */
 		cpufreq_verify_within_limits(policy, policy->cpuinfo.min_freq,
-					K1_MAX_FREQ_LIMITATION);
+					policy->cpuinfo.max_freq);
 	}
 
 	cpufreq_for_each_valid_entry(pos, policy->freq_table) {
@@ -471,8 +474,8 @@ void spacemit_cpufreq_ready(struct cpufr
 		/* M1 */
 	} else {
 		/* K1 or other */
-		remove_policy_boost_sysfs_file(policy);
-		remove_boost_sysfs_file();
+		/* remove_policy_boost_sysfs_file(policy);
+		 * remove_boost_sysfs_file(); */
 	}
 }
 
